pkg/scalers/redis_scaler.go

```diff
@@ -197,9 +197,30 @@ func getRedisListLength(ctx context.Context, address string, password string, li
 	}
 
 	client := redis.NewClient(options)
+	var listType *redis.StatusCmd
 
-	cmd := client.LLen(listName)
+	listType = client.Type(listName)
+	if listType.Err() != nil {
+		return -1, listType.Err()
+	}
+
+	var cmd *redis.IntCmd
+	switch listType.Val() {
```

Comment:
Do we just pull from the correct type or does the type have to be configured?

Reply:
The way I'm reading the change, _and I'm no redis expert btw_, is that the particular list we're trying to scale on could have different types that require a different call on the client. I think the approach looks fine to me.

Reply:
> Do we just pull from the correct type or does the type have to be configured?

it may be configured by the user but we sill will have to do switch casing on it for different api call to redis, so this approach is more flexible to the user

Reply:
If it's a type of list, then it's ok for me but if it's a different type of entity I would make it configurable on what people want to get

Reply:
lets call it collection instead of a list, and you have different type of collections(list, set, map, etc ..) that all have "name" and "length" and those are the only parameters i would want the user to configure